src/Admin/Admin.php

```diff
@@ -230,4 +238,59 @@ public function get_view( string $view, array $context_variables = [] ): string
 	protected function enableReports(): bool {
 		return apply_filters( 'woocommerce_gla_enable_reports', true );
 	}
+
+	/**
+	 * This method is ONLY used during development.
+	 *
+	 * The runtime.js file is created when the front-end is developed in Fast Refresh mode
+	 * and must be loaded together to enable the mode.
+	 *
+	 * When Gutenberg is not installed or not activated, the react dependency will not have
+	 * the 'wp-react-refresh-entry' handle, so here injects the Fast Refresh scripts we built.
+	 *
+	 * The Fast Refresh also needs the development version of React and ReactDOM.
+	 * They will be replaced if the SCRIPT_DEBUG flag is not enabled.
+	 *
+	 * @param WP_Scripts $scripts WP_Scripts instance.
+	 */
+	private function inject_fast_refresh_for_dev( $scripts ) {
+		if ( ! file_exists( "{$this->get_root_dir()}/js/build/runtime.js" ) ) {
```

Comment:
❓  Why here we use `{$this->get_root_dir()}/js/build/runtime.js`. but in line 277 we use `{$plugin_url}/js/build/runtime.js` it's a different file? or just two ways to get the same thing?

Reply:
The `file_exists` checks whether a file exists via the file path in the filesystem, and the enqueue script function will let the browser load the script file via the server URL. So they are the same file but need different URIs.